### ISSUE TITLE
[BUG] Error "r(3000)" when running reghdfe and ppmlhdfe

### ISSUE THREAD

**Issue by polinaust** [sergiocorreia/reghdfe#237]
# Before submitting the bug report

1. Verify that you are using the latest versions of reghdfe and ftools (`which reghdfe`). Note that the latest version is usually on Github and not on SSC.
. which ftools
c:\ado\plus\f\ftools.ado
*! version 2.37.0 16aug2019

. which reghdfe
c:\ado\plus\r\reghdfe.ado
*! version 6.12.1 27June2021

2. Verify that your Stata is updated (`update query`).
I use Stata 17 installed on a server without internet connection.

# Bug report

**Please complete the following information:**

- Stata version: 17.0
- OS: Windows Server

**Behavior**

I've used both reghdfe and ppmlhdfe quite frequently in the last few days, and they work without any problems. However, today when I tried to run regressions none of the commands work.

Here is an example for reghdfe but I receive the same error when I use ppmlhdfe as well:
. reghdfe count i.treat##i.post2013, a(year country_id) cluster(techn_code)
function cleanup_before_saving() not declared in class Factor
(57 lines skipped)
(error occurred while loading reghdfe.ado)
r(3000);

The same error even here:
. reghdfe, version
function cleanup_before_saving() not declared in class Factor
(57 lines skipped)
(error occurred while loading reghdfe.ado)
r(3000);

I've tried some simple things like restarting Stata but it doesn't help.

I have both packages installed on my local computer (Stata 15.0) with internet connection, and everything works perfectly well. So, the problem only occurs when I use ppmlhdfe or reghdfe with Stata installed on a server. 

What can I do to try to solve the problem?

**Comment by TingjuHsu** [comment]
Hi @polinaust , I encountered the same issue today. My ftools is the same version with reghdfe version a little later 6.12.2 dated 02Nov2021. Did you solve your problem? If so, can you share how did you solve it?
Thanks!

**Comment by sergiocorreia** [comment]
The latest version of ftools is 2.48.8: https://github.com/sergiocorreia/ftools/releases/tag/2.48.0 Can you try with the updated version?

**Comment by TingjuHsu** [comment]
I will try. Thanks!
Ting

**Comment by polinau** [comment]
@TingjuHsu, we did solve the problem, and now both reghdfe and ppmlhdfe work properly. It was some time ago, and I'm not sure what measures exactly helped to solve the issue. I think what we did was that we deleted completely all related packages from the server and installed the latest versions from GitHub. Because the server where Stata is installed doesn't have an internet connection I downloaded manually all necessary files from GitHub, followed all the instructions on manual install from here https://github.com/sergiocorreia/reghdfe, and then those packages were added to Stata on the server.
